frame/election-provider-multi-phase/src/weights.rs

```diff
@@ -69,86 +69,61 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	// Storage: ElectionProviderMultiPhase CurrentPhase (r:0 w:1)
 	fn on_initialize_open_signed() -> Weight {
-		(107_551_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(10 as Weight))
-			.saturating_add(T::DbWeight::get().writes(4 as Weight))
-	}
-	// Storage: Staking CounterForValidators (r:1 w:0)
-	// Storage: Staking Validators (r:2 w:0)
-	// Storage: Staking CounterForNominators (r:1 w:0)
-	// Storage: Staking SlashingSpans (r:1 w:0)
-	// Storage: Staking Bonded (r:1 w:0)
-	// Storage: Staking Ledger (r:1 w:0)
-	// Storage: Staking Nominators (r:1 w:0)
-	// Storage: Staking ValidatorCount (r:1 w:0)
-	// Storage: ElectionProviderMultiPhase Round (r:1 w:0)
-	// Storage: ElectionProviderMultiPhase SnapshotMetadata (r:0 w:1)
-	// Storage: ElectionProviderMultiPhase DesiredTargets (r:0 w:1)
-	// Storage: ElectionProviderMultiPhase Snapshot (r:0 w:1)
-	// Storage: ElectionProviderMultiPhase CurrentPhase (r:0 w:1)
-	fn on_initialize_open_unsigned_with_snapshot() -> Weight {
-		(96_899_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(10 as Weight))
-			.saturating_add(T::DbWeight::get().writes(4 as Weight))
+		(30_788_000 as Weight)
+			.saturating_add(T::DbWeight::get().reads(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	// Storage: ElectionProviderMultiPhase Round (r:1 w:0)
 	// Storage: ElectionProviderMultiPhase CurrentPhase (r:0 w:1)
-	fn on_initialize_open_unsigned_without_snapshot() -> Weight {
-		(18_549_000 as Weight)
+	fn on_initialize_open_unsigned() -> Weight {
+		(29_876_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	// Storage: System Account (r:1 w:1)
 	// Storage: ElectionProviderMultiPhase QueuedSolution (r:0 w:1)
 	fn finalize_signed_phase_accept_solution() -> Weight {
-		(48_349_000 as Weight)
+		(48_776_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(2 as Weight))
 	}
 	// Storage: System Account (r:1 w:1)
 	fn finalize_signed_phase_reject_solution() -> Weight {
-		(32_014_000 as Weight)
+		(32_575_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
+	// Storage: ElectionProviderMultiPhase SnapshotMetadata (r:0 w:1)
+	// Storage: ElectionProviderMultiPhase DesiredTargets (r:0 w:1)
+	// Storage: ElectionProviderMultiPhase Snapshot (r:0 w:1)
+	fn create_snapshot_internal() -> Weight {
+		(8_863_521_000 as Weight)
+			.saturating_add(T::DbWeight::get().writes(3 as Weight))
+	}
 	// Storage: ElectionProviderMultiPhase SignedSubmissionIndices (r:1 w:1)
 	// Storage: ElectionProviderMultiPhase SignedSubmissionNextIndex (r:1 w:1)
 	// Storage: ElectionProviderMultiPhase SnapshotMetadata (r:1 w:1)
 	// Storage: ElectionProviderMultiPhase SignedSubmissionsMap (r:1 w:0)
+	// Storage: System BlockWeight (r:1 w:1)
```

Comment:
I assume this is coming from `T::BlockWeights::get()`?

Reply:
yes

Reply:
oh no -- actually this should be because we're manually writing to the weight storage, and my assumption was that this is whitelisted. Have to check this one ^^

Reply:
@shawntabrizi I am surprised that this is not whitelisted, you agree that it should be? every transaction will need to touch that in a block.